app/assets/javascripts/js-routes.js.erb

```diff
@@ -1,2 +1,3 @@
 <%# encoding: UTF-8 %>
+<%= depend_on "file://#{Rails.root.join 'config', 'routes.rb'}" %>
```

Comment:
What is the outcome of `depend_on`? What does it write the JS file?
I saw sprockets describes `depend_on` as a directive. What is the difference of using it like ruby method instead of `// depend_on` sprocket directive?